dot/build_spec_test.go

```diff
@@ -0,0 +1,301 @@
+    "telemetryEndpoints": null,
+    "protocolId": "protocol",
+    "genesis": {},
+    "properties": {
+        "key": "value"
+    },
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tt.buildSpec.ToJSON()
+			assert.ErrorIs(t, err, tt.err)
+			assert.Equal(t, strings.TrimSpace(tt.want), string(got))
+		})
+	}
+}
+
+func TestBuildFromDB(t *testing.T) {
+	// initialise node (initialise state database and load genesis data)
+	cfg := NewTestConfig(t)
+	cfg.Init.Genesis = utils.GetGssmrGenesisRawPathTest(t)
+	builder := nodeBuilder{}
+	err := builder.initNode(cfg)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name string
+		path string
+		want *BuildSpec
+		err  error
+	}{
+		{name: "normal conditions", path: cfg.Global.BasePath,
+			want: &BuildSpec{genesis: &genesis.Genesis{
+				Name:       "Gossamer",
+				ID:         "gssmr",
+				Bootnodes:  []string{},
+				ProtocolID: "/gossamer/gssmr/0",
+				Genesis: genesis.Fields{
+					Raw:     map[string]map[string]string{},
+					Runtime: map[string]map[string]interface{}{},
+				},
+			}}},
+		{name: "invalid db path", path: t.TempDir(),
+			err: errors.New("cannot start state service: failed to create block state: cannot get block 0: Key not found")},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := BuildFromDB(tt.path)
+
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			if tt.want != nil {
+				got.genesis.Genesis.Raw = map[string]map[string]string{}
+				got.genesis.Genesis.Runtime = map[string]map[string]interface{}{}
+				assert.Equal(t, tt.want, got)
+			}
+		})
+	}
+}
+
+func TestBuildFromGenesis(t *testing.T) {
+	testGenesisPath := genesis.CreateTestGenesisJSONFile(t, false)
+
+	type args struct {
+		path      string
+		authCount int
+	}
+	tests := []struct {
+		name string
+		args args
+		want *BuildSpec
+		err  error
+	}{
+		{
+			name: "invalid file path",
+			args: args{
+				path: "/invalid/path",
+			},
+			err: errors.New("open /invalid/path: no such file or directory"),
+		},
+		{
+			name: "normal conditions",
+			args: args{
+				path: testGenesisPath,
+			},
+			want: &BuildSpec{genesis: &genesis.Genesis{
+				Name: "test",
+				Genesis: genesis.Fields{
+					Raw: map[string]map[string]string{"top" +
+						"": {"0x26aa394eea5630e07c48ae0c9558cef7c21aab032aaa6e946ca50ad39ab66603": "0x01",
+						"0x3a636f6465": "mocktestcode"}},
+					Runtime: map[string]map[string]interface{}{"System": {"code": "mocktestcode"}},
+				},
+			}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := BuildFromGenesis(tt.args.path, tt.args.authCount)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			if tt.want != nil {
```

Comment:
check not required, look at the comment above

Reply:
Removed check.